Intermittent test failure when testing tail streaming

```
<nil>
&{{0 0 <nil>} {0 0 <nil>} false  <nil> 0xc00044d540 0x554060 0 0 0xc00027cf60 16384 test 0xc0002d6f20 0xc0004d4980 false {{} {{} {} 0} 0} 0xc0002eee70 0xc0002eeee0 0xc0002eef50 <nil> {{} {0 0}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}}}
--- FAIL: TestStreamTailWithFollow (0.02s)
    --- FAIL: TestStreamTailWithFollow/with_follow_data,_past_data_arrives_after (0.00s)
        stream_test.go:948: 
            	Error Trace:	/home/runner/work/kubetail/kubetail/modules/shared/logs/stream_test.go:948
            	Error:      	Not equal: 
            	            	expected: []string{"s1-a", "s2-a", "s1-b", "s2-b"}
            	            	actual  : []string{"s1-a", "s2-a", "s2-b", "s1-b"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -3,4 +3,4 @@
            	            	  (string) (len=4) "s2-a",
            	            	- (string) (len=4) "s1-b",
            	            	- (string) (len=4) "s2-b"
            	            	+ (string) (len=4) "s2-b",
            	            	+ (string) (len=4) "s1-b"
            	            	 }
            	Test:       	TestStreamTailWithFollow/with_follow_data,_past_data_arrives_after
```

--- a/modules/shared/logs/stream_test.go
+++ b/modules/shared/logs/stream_test.go
@@ -515,7 +515,8 @@ func TestStreamAllWithFollow(t *testing.T) {
 		setPastStreamAfter1  []LogRecord
 		setPastStreamAfter2  []LogRecord
 		setFutureStream      []LogRecord
-		wantLines            []string
+		wantPastLines        []string
+		wantFutureLines      []string
 	}{
 		{
 			"no follow data, past data arrives before",
@@ -525,6 +526,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]string{"s1-a", "s2-a"},
+			[]string{},
 		},
 		{
 			"no follow data, past data arrives after",
@@ -534,6 +536,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{logs2[0]},
 			[]LogRecord{},
 			[]string{"s1-a", "s2-a"},
+			[]string{},
 		},
 		{
 			"no follow data, past data arrives before and after",
@@ -543,6 +546,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{logs2[1]},
 			[]LogRecord{},
 			[]string{"s1-a", "s2-a", "s1-b", "s2-b"},
+			[]string{},
 		},
 		{
 			"with follow data, past data arrives before",
@@ -551,7 +555,8 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]LogRecord{logs1[1]},
-			[]string{"s1-a", "s2-a", "s1-b"},
+			[]string{"s1-a", "s2-a"},
+			[]string{"s1-b"},
 		},
 		{
 			"with follow data, past data arrives after",
@@ -560,7 +565,8 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{logs1[0]},
 			[]LogRecord{logs2[0]},
 			[]LogRecord{logs1[1], logs2[1]},
-			[]string{"s1-a", "s2-a", "s1-b", "s2-b"},
+			[]string{"s1-a", "s2-a"},
+			[]string{"s1-b", "s2-b"},
 		},
 		{
 			"with follow data, past data arrives before and after",
@@ -569,7 +575,8 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{logs1[1]},
 			[]LogRecord{logs2[1]},
 			[]LogRecord{logs1[2], logs2[2]},
-			[]string{"s1-a", "s2-a", "s1-b", "s2-b", "s1-c", "s2-c"},
+			[]string{"s1-a", "s2-a", "s1-b", "s2-b"},
+			[]string{"s1-c", "s2-c"},
 		},
 		{
 			"with follow data, no past data",
@@ -578,6 +585,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]LogRecord{logs1[0], logs1[1]},
+			[]string{},
 			[]string{"s1-a", "s1-b"},
 		},
 	}
@@ -664,6 +672,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 
 			// Get log records in goroutine
 			messages := []string{}
+			expectedTotal := len(tt.wantPastLines) + len(tt.wantFutureLines)
 
 			doneCh := make(chan struct{})
 			go func() {
@@ -673,7 +682,7 @@ func TestStreamAllWithFollow(t *testing.T) {
 					messages = append(messages, r.Message)
 
 					// Exit after expected number of messages arrives
-					if len(messages) == len(tt.wantLines) {
+					if len(messages) == expectedTotal {
 						break
 					}
 				}
@@ -708,8 +717,14 @@ func TestStreamAllWithFollow(t *testing.T) {
 			// Wait for all messages to arrive
 			<-doneCh
 
-			// Check result
-			assert.Equal(t, tt.wantLines, messages)
+			// Check result: past data should be in strict order, future data can be in any order
+			numPast := len(tt.wantPastLines)
+			if numPast > 0 {
+				assert.Equal(t, tt.wantPastLines, messages[:numPast], "past data should be in strict order")
+			}
+			if len(tt.wantFutureLines) > 0 {
+				assert.ElementsMatch(t, tt.wantFutureLines, messages[numPast:], "future data should contain all expected records (order doesn't matter)")
+			}
 		})
 	}
 }
@@ -745,7 +760,8 @@ func TestStreamTailWithFollow(t *testing.T) {
 		setPastStreamAfter1  []LogRecord
 		setPastStreamAfter2  []LogRecord
 		setFutureStream      []LogRecord
-		wantLines            []string
+		wantPastLines        []string
+		wantFutureLines      []string
 	}{
 		{
 			"no follow data, past data arrives before",
@@ -756,6 +772,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]string{"s1-b", "s2-b"},
+			[]string{},
 		},
 		{
 			"no follow data, past data arrives after",
@@ -766,6 +783,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{logs2[0], logs2[1]},
 			[]LogRecord{},
 			[]string{"s1-b", "s2-b"},
+			[]string{},
 		},
 		{
 			"no follow data, past data arrives before and after",
@@ -776,6 +794,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{logs2[0]},
 			[]LogRecord{},
 			[]string{"s1-b", "s2-b"},
+			[]string{},
 		},
 		{
 			"with follow data, past data arrives before",
@@ -785,7 +804,8 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]LogRecord{logs1[1]},
-			[]string{"s1-a", "s2-a", "s1-b"},
+			[]string{"s1-a", "s2-a"},
+			[]string{"s1-b"},
 		},
 		{
 			"with follow data, past data arrives after",
@@ -795,7 +815,8 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{logs1[0]},
 			[]LogRecord{logs2[0]},
 			[]LogRecord{logs1[1], logs2[1]},
-			[]string{"s1-a", "s2-a", "s1-b", "s2-b"},
+			[]string{"s1-a", "s2-a"},
+			[]string{"s1-b", "s2-b"},
 		},
 		{
 			"with follow data, past data arrives before and after",
@@ -805,7 +826,8 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{logs1[0]},
 			[]LogRecord{logs2[0]},
 			[]LogRecord{logs1[2], logs2[2]},
-			[]string{"s1-b", "s2-b", "s1-c", "s2-c"},
+			[]string{"s1-b", "s2-b"},
+			[]string{"s1-c", "s2-c"},
 		},
 		{
 			"with follow data, no past data",
@@ -815,6 +837,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 			[]LogRecord{},
 			[]LogRecord{},
 			[]LogRecord{logs1[0], logs1[1]},
+			[]string{},
 			[]string{"s1-a", "s1-b"},
 		},
 	}
@@ -900,6 +923,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 
 			// Get log records in goroutine
 			messages := []string{}
+			expectedTotal := len(tt.wantPastLines) + len(tt.wantFutureLines)
 
 			doneCh := make(chan struct{})
 			go func() {
@@ -909,7 +933,7 @@ func TestStreamTailWithFollow(t *testing.T) {
 					messages = append(messages, r.Message)
 
 					// Exit after expected number of messages arrives
-					if len(messages) == len(tt.wantLines) {
+					if len(messages) == expectedTotal {
 						break
 					}
 				}
@@ -944,8 +968,14 @@ func TestStreamTailWithFollow(t *testing.T) {
 			// Wait for all messages to arrive
 			<-doneCh
 
-			// Check result
-			assert.Equal(t, tt.wantLines, messages)
+			// Check result: past data should be in strict order, future data can be in any order
+			numPast := len(tt.wantPastLines)
+			if numPast > 0 {
+				assert.Equal(t, tt.wantPastLines, messages[:numPast], "past data should be in strict order")
+			}
+			if len(tt.wantFutureLines) > 0 {
+				assert.ElementsMatch(t, tt.wantFutureLines, messages[numPast:], "future data should contain all expected records (order doesn't matter)")
+			}
 		})
 	}
 }

